clang/lib/Lex/PPMacroExpansion.cpp

```diff
@@ -1722,10 +1722,12 @@ void Preprocessor::ExpandBuiltinMacro(Token &Tok) {
     // MSVC, ICC, GCC, VisualAge C++ extension.  The generated string should be
     // of the form "Ddd Mmm dd hh::mm::ss yyyy", which is returned by asctime.
     const char *Result;
+    char TimeString[std::size("Ddd Mmm dd hh:mm:ss yyyy")];
     if (getPreprocessorOpts().SourceDateEpoch) {
       time_t TT = *getPreprocessorOpts().SourceDateEpoch;
       std::tm *TM = std::gmtime(&TT);
```

Comment:
As an aside, I find it weird that we use gmtime in one case and localtime in the other (see L1739)

Reply:
It's consistent with the behavior in `ComputeDATE_TIME`, perhaps because `SourceDateEpoch` is the number of seconds since the epoch and it's easier to reason about that from UTC as a base than localtime (e.g. for build reproducibility)